xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -375,16 +374,23 @@ public void sendMessage(String message) {
                     Subscription newSubscription = new Subscription(e, channelName, args);
                     try {
                       sendMessage(getSubscribeMessage(channelName, args));
-                    } catch (IOException throwable) {
-                      e.onError(throwable);
+                    } catch (
+                        Exception
+                            throwable) { // if getSubscribeMessage throws this, it is because it
+                      // needs to report
+                      e.onError(throwable); // a problem creating the message
                     }
                     return newSubscription;
                   });
             })
         .doOnDispose(
             () -> {
               if (channels.remove(channelId) != null) {
-                sendMessage(getUnsubscribeMessage(channelId));
+                try {
+                  sendMessage(getUnsubscribeMessage(channelId));
+                } catch (IOException e) {
+                  LOG.error("Failed to unsubscribe channel: {}", channelId);
```

Comment:
Do we want to log more information about what the error that occurred here was?

Reply:
Well do we really care why it failed to unsubscribe on an IOException?  I assume the socket has gone anyway, and the message (should be warn) is just to avoid eating the exception.

Reply:
Probably not, but in some cases this information may yield more color as to what is going on, figured it made sense to consider.

Reply:
Hm, on principle, eating an unknown exception without a stack trace isn't the best idea IMO.   If you know what it is and want to suppress noise, fine, but as you say below, this could be caused by anything.  I should have spotted this, sorry.